bson/son.py

```diff
@@ -176,14 +183,16 @@ def transform_value(value):
 
         return transform_value(dict(self))
 
-    def __deepcopy__(self, memo):
-        out = SON()
+    def __deepcopy__(self, memo: MutableMapping[int, "SON[_Key, _Value]"]) -> "SON[_Key, _Value]":
+        out: SON[_Key, _Value] = SON()
         val_id = id(self)
         if val_id in memo:
-            return memo.get(val_id)
+            value = memo.get(val_id)
+            assert value is not None
+            return value
```

Comment:
Ideally `if val_id in memo` would guard against the `get()` being `None`, but it does not, which is why I added the intermediate variable and guard.

Reply:
Yeah I agree but I think it's similar to the other issue since mypy can't determine if the val_id might be removed by another thread in between the if-statement and the get() call. Regardless, I think we can replace these 3 lines with `return memo[val_id]` since we know val_id is always there. Then we don't need to add redundant code just to appease mypy.

Reply:
Used type ignore instead